added ubuntu box

d78ui98 opened the pull request:
Installing LTSP on Ubuntu 18.04 using vagrant

- [x] Network setup
- [x] Shell provisioner to install ltsp
- [x] setup up configuration files

fixes #5 #9

d78ui98 committed:
added ubuntu ltsp box

d78ui98 committed:
using ubuntu box with vagrant

d78ui98 committed:
adding bridged adapter

d78ui98 committed:
added adapter name

akash0x53 commented:
Try to follow this git commit message guidelines by Linus - these are just guidelines.
 

```
A good commit message looks like this:

	Header line: explaining the commit in one line

	Body of commit message is a few lines of text, explaining things
	in more detail, possibly giving some background about the issue
	being fixed, etc etc.

	The body of the commit message can be several paragraphs, and
	please do proper word-wrap and keep columns shorter than about
	74 characters or so. That way "git log" will show things
	nicely even when it's indented.

	Reported-by: whoever-reported-it
	Signed-off-by: Your Name <youremail@yourhost.com>

where that header line really should be meaningful, and really should be
just one line.  That header line is what is shown by tools like gitk and
shortlog, and should summarize the change in one readable line of text,
independently of the longer explanation.
```

d78ui98 commented:
>Since we are making the changes on a branch, when we merge it we can squash all the commits in one

Yes I was going to that only. I am following your instructions from [here](https://github.com/orgs/docker-scripts/teams/ltsp/discussions/9)  at point 7 :)

dashohoxha commented:
> I am following your instructions from here at point 7 :)

But still it is a good idea to test your changes locally, before committing and pushing them, if this is possible. There may be some cases when this is difficult though, so feel free to do what you think is best..

dashohoxha commented:
@d78ui98 I think that you should commit your changes at the end of each day, even if they are not yet consistent or well tested. This is useful so that we can see your progress.

d78ui98 commented:
Okay I will do that.

d78ui98 committed:
added shell provisioner

I have enabled the promiscous mode fixing spaceing and added the shell script to install ltsp to vagrant box.

d78ui98 commented:
adding `xenial` and `bionic` box was a good idea. Vagrantfile ran smoothly in fact it was even faster than `trusty`. I was thinking to adding all the 3 Vagrantfiles seperately in different folders. This would allow fexiblity to users to choose which box to run. Let me know if I should do that.

akash0x53 commented:
Vagrantfile is basically ruby code. You can try to change box dynamically instead maintaining different files - if at all needed to have different Ubuntu flavors.

dashohoxha commented:
> I was thinking to adding all the 3 Vagrantfiles seperately in different folders. This would allow fexiblity to users to choose which box to run.

I think that you should use only the latest stable version. For Ubuntu this is `bionic`. If everything works  fine with bionic, forget about the rest.

Then, for creating a Debian version, I suggest that we use different git branches. This means that we will have a branch called `bionic` with the corresponding Vagrantfiles, shell scripts, etc. and another branch called `buster` (or `stretch`) with the corresponding Debian version and scripts. Something similar to this repo: https://github.com/EasyGnuPG/ggpg-ds

Another solution could be to create different github repositories, like `ltsp-bionic` and `ltsp-buster`.

dashohoxha commented:
> Vagrantfile is basically ruby code. You can try to change box dynamically instead maintaining different files - if at all needed to have different Ubuntu flavors.

This would be OK if the provisioning bash scripts were the same. But this may not be the case, especially when switching from ubuntu to debian. So it may be better to keep different versions of the Vagrantfile and bash scripts.

d78ui98 committed:
vagrantfile for bionic

I have added Vagrantfile for ubuntu bionic. tested it and it is working great. Also added some extra packages required and comments to the provisioner bash script.

d78ui98 committed:
bridged mode to eno1

dashohoxha commented on ubuntu LTSP box/Vagrantfile:
```diff
@@ -19,18 +19,45 @@ end
 
 $LTSP_Installer_script = <<SCRIPT
 #!/bin/bash
```
You can also separate the bash script from the Vagrantfile, like this example: https://github.com/Own-Mailbox/omb/blob/debian/Vagrantfile#L14
This would make more readable both the Vagrantfile and the installation/configuration script.

d78ui98 replied:
okay that would be better. Done.

d78ui98 committed:
added external script

replaced the inline script with external script to provision the host machine

d78ui98 committed:
eno1

d78ui98 commented:
@dashohoxha @akash0x53 please suggest any other improvements that I should make.

dashohoxha commented:
@d78ui98 have you tested it? Does it work? If yes, we can merge it and do any improvements after that.
One thing that is missing is that you have not installed ltsp-manager. But you can do it after merging this branch.

d78ui98 commented:
Yes I have tested it multiple times. It works great. 
>One thing that is missing is that you have not installed ltsp-manager. But you can do it after merging this branch.

Sure I will install ltsp-manager after its merged.

dashohoxha commented:
I merged it and moved it to the branch `bionic`. This is the branch where you should work from now on.
I also created the branch `buster` for the Debian version.
